lib/insights_cloud/async/cloud_connector_announce_task.rb

```diff
@@ -0,0 +1,47 @@
+module InsightsCloud
+  module Async
+    class CloudConnectorAnnounceTask < ::Actions::EntryAction
+      def self.subscribe
+        Actions::RemoteExecution::RunHostsJob
+      end
+
+      def self.connector_feature_id
+        @connector_feature_id ||= RemoteExecutionFeature.feature!(ForemanRhCloud::CloudConnector::CLOUD_CONNECTOR_FEATURE).id
+      end
+
+      def plan(job_invocation)
+        return unless connector_playbook_job?(job_invocation)
+
+        plan_self
+      end
+
+      def run(event = nil)
+        # Handle skip events
+        return if event == Dynflow::Action::Skip
```

Comment:
You could make `#run` accept no arguments and then you could drop this line and it should behave exactly the same way